bundles/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/internal/handler/OceanicThingHandler.java

```diff
@@ -103,18 +103,19 @@ public void initialize() {
         } else {
             bufferSize = ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue();
         }
-
+        ScheduledFuture<?> pollingJob = this.pollingJob;
         if (pollingJob == null || pollingJob.isCancelled()) {
-            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1,
+            this.pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1,
                     ((BigDecimal) getConfig().get(INTERVAL)).intValue(), TimeUnit.SECONDS);
         }
     }
 
     @Override
     public void dispose() {
+        ScheduledFuture<?> pollingJob = this.pollingJob;
```

Comment:
Optional/additional - we could also skip the `!pollingJob.isCancelled()` check below.

Reply:
thnx